Sources/RabbitMq/Publisher/Publisher.swift

```diff
@@ -12,10 +12,15 @@ import NIOCore
 /// )
 /// try await publisher.publish("Publish this!")
 /// ```
-public struct Publisher: Sendable {
+public struct Publisher: ConnectionLoggable {
     let connection: Connection
     let configuration: PublisherConfiguration
-    let logger: Logger
+
+    var logger: Logger {
+        get async {
+            await connection.logger.withMetadata(["exchangeName": .string(configuration.exchangeName)])
```

Comment:
This creates a new logger with metadata every time `logger` is used. This is done even when the log is not actually created due to the current log level.